File: SmartPhone++/SmartPhone.Services/Database/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Services.Database
{
    public class Order
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string OrderNumber { get; set; } = string.Empty;
        
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        
        public DateTime? ShippedDate { get; set; }
        
        public DateTime? DeliveredDate { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal Subtotal { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal TaxAmount { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal ShippingAmount { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal DiscountAmount { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal TotalAmount { get; set; }
        
        [Required]
        [MaxLength(3)]
        public string CurrencyCode { get; set; } = string.Empty; // USD, EUR, BAM, etc.
        
        [Required]
        [MaxLength(5)]
        public string CurrencySymbol { get; set; } = string.Empty; // $, €, KM
        
        [MaxLength(50)]
        public string Status { get; set; } = "Pending"; // Pending, Processing, Shipped, Delivered, Cancelled
        
        [MaxLength(500)]
        public string? Notes { get; set; }
        
        [MaxLength(100)]
        public string? TrackingNumber { get; set; }
        
        // Shipping information
        [Required]
        [MaxLength(100)]
        public string ShippingFirstName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ShippingLastName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string ShippingAddress { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ShippingCity { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(20)]
        public string ShippingPostalCode { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ShippingCountry { get; set; } = string.Empty;
        
        [Required]
        [Phone]
        [MaxLength(20)]
        public string ShippingPhone { get; set; } = string.Empty;
        
        [EmailAddress]
        [MaxLength(100)]
        public string? ShippingEmail { get; set; }
        
        // Billing information
        [Required]
        [MaxLength(100)]
        public string BillingFirstName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string BillingLastName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string BillingAddress { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string BillingCity { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(20)]
        public string BillingPostalCode { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string BillingCountry { get; set; } = string.Empty;
        
        [Required]
        [Phone]
        [MaxLength(20)]
        public string BillingPhone { get; set; } = string.Empty;
        
        [EmailAddress]
        [MaxLength(100)]
        public string? BillingEmail { get; set; }
        
        // Foreign keys
        public int UserId { get; set; }
        public int? CouponId { get; set; }
        
        // Navigation properties
        public User User { get; set; } = null!;
        public Coupon? Coupon { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        public ICollection<OrderStatusHistory> OrderStatusHistory { get; set; } = new List<OrderStatusHistory>();
    }
} 

File: SmartPhone++/UI/smartphone_mobile_client/lib/model/order.dart
import 'package:json_annotation/json_annotation.dart';
import 'order_item.dart';

part 'order.g.dart';

@JsonSerializable()
class Order {
  final int id;
  final String orderNumber;
  final DateTime orderDate;
  final DateTime? shippedDate;
  final DateTime? deliveredDate;
  final double subtotal;
  final double taxAmount;
  final double shippingAmount;
  final double discountAmount;
  final double totalAmount;
  final String status;
  final String? notes;
  final String? trackingNumber;
  
  // Shipping information
  final String shippingFirstName;
  final String shippingLastName;
  final String shippingAddress;
  final String shippingCity;
  final String shippingPostalCode;
  final String shippingCountry;
  final String shippingPhone;
  final String? shippingEmail;
  
  // Billing information
  final String billingFirstName;
  final String billingLastName;
  final String billingAddress;
  final String billingCity;
  final String billingPostalCode;
  final String billingCountry;
  final String billingPhone;
  final String? billingEmail;
  
  final int userId;
  final String userName;
  final List<OrderItem>? orderItems;

  Order({
    required this.id,
    required this.orderNumber,
    required this.orderDate,
    this.shippedDate,
    this.deliveredDate,
    required this.subtotal,
    required this.taxAmount,
    required this.shippingAmount,
    required this.discountAmount,
    required this.totalAmount,
    required this.status,
    this.notes,
    this.trackingNumber,
    required this.shippingFirstName,
    required this.shippingLastName,
    required this.shippingAddress,
    required this.shippingCity,
    required this.shippingPostalCode,
    required this.shippingCountry,
    required this.shippingPhone,
    this.shippingEmail,
    required this.billingFirstName,
    required this.billingLastName,
    required this.billingAddress,
    required this.billingCity,
    required this.billingPostalCode,
    required this.billingCountry,
    required this.billingPhone,
    this.billingEmail,
    required this.userId,
    required this.userName,
    this.orderItems,
  });

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
  Map<String, dynamic> toJson() => _$OrderToJson(this);

  String get fullShippingName => '$shippingFirstName $shippingLastName';
  String get fullBillingName => '$billingFirstName $billingLastName';
  String get formattedOrderDate => '${orderDate.day}/${orderDate.month}/${orderDate.year}';
  String get formattedTotalAmount => '${totalAmount.toStringAsFixed(2)} BAM';
  
  bool get isShipped => status == 'Shipped';
  bool get isDelivered => status == 'Delivered';
  bool get isProcessing => status == 'Processing';
  bool get isPending => status == 'Pending';
  bool get isCancelled => status == 'Cancelled';
}
